packages/hasura/src/dataProvider/index.ts

```diff
@@ -38,6 +43,36 @@ const dataProvider = (
                 ? `${meta?.operation ?? resource}_by_pk`
                 : camelCase(`${meta?.operation ?? resource}_by_pk`);
 
+            const gqlOperation = meta?.gqlQuery ?? meta?.gqlMutation;
+
+            if (gqlOperation) {
+                let query = gqlOperation;
+                const variables = {
+                    id,
+                };
+
+                if (isMutation(gqlOperation)) {
+                    const stringFields = getOperationFields(gqlOperation);
+
+                    query = gqlTag`
+                        query Get${operation}($id: ${getIdType(resource)}!) {
```

Comment:
```suggestion
                        query Get${xxxxx}($id: ${getIdType(resource)}!) {
```

This one generates incorrect operation name: `query Getposts_by_pk($id: uuid!) {`

We should use PascalCased resource name.